code/parse/sexp.cpp

```diff
@@ -23586,7 +23567,7 @@ int eval_sexp(int cur_node, int referenced_node)
 
 			case OP_DO_FOR_VALID_ARGUMENTS:
 				// do-for-valid-arguments should only ever be called within eval_when()
-				Int3();
+				Assertion(false, "do-for-valid-arguments was encountered in eval_sexp()!");
```

Comment:
This should be changed to a `Warning`. It is admittedly a violation of the programming contract, but at the FREDder level, not the game engine level. A FREDder could conceivably make this mistake, and a `Warning` is more FREDder-friendly.